### PR TITLE
bpo-30052: DOC: Link `bytes` to stdtypes not functions 

### PR DESCRIPTION
**Pull request by csabella**
Change for :class:`bytes` and :class:`bytesarray` to link to the stdtypes page instead of the functions page.

Note: to check all the links, I needed to use 'make clean' then 'make html'.  I don't know if that would be done automatically when merging.

### CODE REVIEW

**Review by ncoghlan — APPROVED** [review]
This looks like a good fix for the original issue to me (as it brings bytes/bytearray cross-references into line with the other builtin container types).

I think we should merge this to dev and backport as far as 3.6, but will leave it open for a day or so to give others a chance to comment.

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -2275,69 +2275,71 @@ binary protocols are based on the ASCII text encoding, bytes objects offer
...
 
-Since 2 hexadecimal digits correspond precisely to a single byte, hexadecimal
-numbers are a commonly used format for describing binary data. Accordingly,
-the bytes type has an additional class method to read data in that format:
+   Since 2 hexadecimal digits correspond precisely to a single byte, hexadecimal
+   numbers are a commonly used format for describing binary data. Accordingly,
+   the bytes type has an additional class method to read data in that format:
 
-.. classmethod:: bytes.fromhex(string)
+   .. classmethod:: bytes.fromhex(string)
```
Comment (berkerpeksag) [inline Doc/library/stdtypes.rst]: `bytes.` prefix is unnecessary now.

File: Doc/library/stdtypes.rst [hunk 2]
```diff
@@ -2275,69 +2275,71 @@ binary protocols are based on the ASCII text encoding, bytes objects offer
...
+         :meth:`bytes.fromhex` now skips all ASCII whitespace in the string,
+         not just spaces.
 
-A reverse conversion function exists to transform a bytes object into its
-hexadecimal representation.
+   A reverse conversion function exists to transform a bytes object into its
+   hexadecimal representation.
 
-.. method:: bytes.hex()
+   .. method:: bytes.hex()
```
Comment (berkerpeksag) [inline Doc/library/stdtypes.rst]: `bytes.` prefix is unnecessary now.

File: Doc/library/stdtypes.rst [hunk 3]
```diff
@@ -2367,49 +2369,53 @@ Bytearray Objects
...
+         not just spaces.
 
-   Return a string object containing two hexadecimal digits for each
-   byte in the instance.
+   A reverse conversion function exists to transform a bytearray object into its
+   hexadecimal representation.
 
-   >>> bytearray(b'\xf0\xf1\xf2').hex()
-   'f0f1f2'
+   .. method:: bytearray.hex()
```
Comment (berkerpeksag) [inline Doc/library/stdtypes.rst]: `bytearray.` prefix is unnecessary now.

File: Doc/reference/datamodel.rst [hunk 4]
```diff
@@ -320,9 +320,9 @@ Sequences
 
          A bytes object is an immutable array.  The items are 8-bit bytes,
          represented by integers in the range 0 <= x < 256.  Bytes literals
-         (like ``b'abc'``) and the built-in function :func:`bytes` can be used to
-         construct bytes objects.  Also, bytes objects can be decoded to strings
-         via the :meth:`~bytes.decode` method.
+         (like ``b'abc'``) and the built-in :func:`bytes()` constructor
```
Comment (berkerpeksag) [inline Doc/reference/datamodel.rst:323]: Why did you add () here? Perhaps using :ref:\`bytes \<func-bytes\>\` would be better?

File: Doc/library/stdtypes.rst (hunk 1 above)
Comment (csabella) [reply Doc/library/stdtypes.rst]: Thank you.  I wasn't sure about these.  Fixed it.

File: Doc/reference/datamodel.rst (hunk 4 above)
Comment (csabella) [reply Doc/reference/datamodel.rst:323]: I added the () to be similar language to set() just below this on the same page.  I had originally changed it to func-butes, but the other sequences on the page didn't refer the the functions.rst, but to the stdtypes.rst, so I erred on the side of keeping them consistent.  Should I change it?

Comment (berkerpeksag) [reply Doc/reference/datamodel.rst:323]: Thanks for the explanation. I'll let @ncoghlan decide which one is more appropriate here. If we decide to go with a reference to functions.rst, the "()" part should be removed (it's redundant, Sphinx will automatically add ()s in the HTML output)

Comment (ncoghlan) [reply Doc/reference/datamodel.rst:323]: This is in the data model section, so referencing the class documentation rather than the builtin function entry makes sense to me.

**Comment by ncoghlan** [comment]
Merged, thanks @csabella!

@berkerpeksag Having merged it, it belatedly occurs to me to ask: do you think we should add a NEWS entry for this? It shouldn't break any incoming links, so it isn't necessary from that perspective, and looking at past Documentation NEWS entries, they seem to mostly be related to documented deprecations, major content additions, and changes to the docs build process.

**Comment by berkerpeksag** [comment]
@ncoghlan +1, a Misc/NEWS entry is not needed in this case.
